spec.md

```diff
@@ -100,6 +110,18 @@ interface ExpressionStatement <: Statement {
 
 An expression statement, i.e., a statement consisting of a single expression.
 
+## Directive
+
+```js
+interface Directive <: ExpressionStatement {
+    directive: string;
+}
+```
+
+A directive from the directive prologue of a script or function.
+The `directive` property is the raw string source of the directive.
+This source includes single or double wrapping quotes and may not contain an EscapeSequence or LineContinuation..
```

Comment:
For non-parsers none of the cases simplifies generation, as both add complexity to AST (but, well, it's evolution). As for semantics - we are not trying to reflect semantics in AST at all, we are reflecting syntax (it's Syntax Tree after all) and thus in many cases map semantically different pieces of programs to identical ASTs.